src/Microsoft.OData.Core/Json/ODataJsonResourceDeserializer.cs

```diff
@@ -986,53 +986,54 @@ internal ODataJsonReaderNestedInfo ReadPropertyWithoutValue(IODataJsonReaderReso
             ODataJsonReaderNestedInfo readerNestedInfo = null;
             IEdmStructuredType resourceType = resourceState.ResourceType;
             IEdmProperty edmProperty = this.ReaderValidator.ValidatePropertyDefined(propertyName, resourceType);
-            if (edmProperty != null && !edmProperty.Type.IsUntyped())
+            if (edmProperty == null || edmProperty.Type.IsUntyped())

```

Comment:
Why has this condition changed?

Reply:
@habbes I meant to leave a comment to explain this change. If you look at `ReadPropertyWithoutValueAsync` method, we invert the condition expression to avoid the unnecessary `if` and `else if`. The logic in the `if (edmProperty == null || edmProperty.Type.IsUntyped())` is what was previously in the `else` statement. The sync method now mirrors the async method and we get rid of unnecessary nesting

Reply:
Had not noticed the inverted if-else blocks.